ports/stm/common-hal/rgbmatrix/RGBMatrix.c

```diff
@@ -27,17 +27,16 @@
 #include <stddef.h>
 
 #include "common-hal/rgbmatrix/RGBMatrix.h"
+#include "timers.h"
 
 #include STM32_HAL_H
 
 extern void _PM_IRQ_HANDLER(void);
 
 void *common_hal_rgbmatrix_timer_allocate() {
-    // TODO(jepler) properly handle resource allocation including never-reset
-    return TIM6;
+    return stm_peripherals_find_timer();
```

Comment:
I think this was just something I didn't pay close enough attention to in the cases you've pointed out, since we never freed the single use timers. You're right of course, I'll double check all instances to make sure there's a proper reserve/free pair per file.